public/app/core/components/colorpicker/withColorPicker.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Drop from 'tether-drop';
+import { SeriesColorPicker } from './SeriesColorPicker';
+
+export interface WithSeriesColorPickerProps {
+  color: string;
+  yaxis?: number;
+  optionalClass?: string;
+  onColorChange: (newColor: string) => void;
+  onToggleAxis?: () => void;
+}
+
+export default function withSeriesColorPicker(WrappedComponent) {
+  return class extends React.Component<WithSeriesColorPickerProps, any> {
+    pickerElem: any;
+    colorPickerDrop: any;
+
+    static defaultProps = {
+      optionalClass: '',
+      yaxis: undefined,
+      onToggleAxis: () => {},
+    };
+
+    constructor(props) {
+      super(props);
+      this.openColorPicker = this.openColorPicker.bind(this);
+    }
+
+    openColorPicker() {
```

Comment:
We usually use arrow functions, that way you don't have to bind it in the constructor.

Reply:
`openColorPicker = () => {}`